infrastructure/storage/src/lmdb.rs

```diff
@@ -0,0 +1,430 @@
+                Ok(())
+            }
+            None => Err(DatastoreError::UnknownDatabase),
+        }
+    }
+
+    fn get_raw(&self, key: &[u8]) -> Result<Option<Vec<u8>>, DatastoreError> {
+        let txn = lmdb::ReadTransaction::new(self.env.clone())?;
+        let accessor = txn.access();
+        match accessor.get::<[u8], [u8]>(&self.curr_db, key).to_opt() {
+            Ok(None) => Ok(None),
+            Ok(Some(v)) => Ok(Some(v.to_vec())),
+            Err(e) => Err(DatastoreError::GetError(format!(
+                "LMDB get error: {}",
+                e.to_string()
+            ))),
+        }
+    }
+
+    fn exists(&self, key: &[u8]) -> Result<bool, DatastoreError> {
+        let txn = lmdb::ReadTransaction::new(self.env.clone())?;
+        let accessor = txn.access();
+        let res: lmdb::error::Result<&lmdb::Ignore> = accessor.get(&self.curr_db, key);
+        Ok(res.to_opt()?.is_some())
+    }
+
+    fn put_raw(&mut self, key: &[u8], value: Vec<u8>) -> Result<(), DatastoreError> {
+        let tx = lmdb::WriteTransaction::new(self.env.clone())?;
+        {
+            let mut accessor = tx.access();
+            accessor.put(&self.curr_db, key, &value, lmdb::put::Flags::empty())?;
+        }
+        tx.commit().map_err(|e| e.into())
+    }
+}
+
+struct LMDBBatch<'a> {
+    db: Arc<lmdb::Database<'static>>,
+    tx: lmdb::WriteTransaction<'a>,
+}
+
+impl<'a> BatchWrite for LMDBBatch<'a> {
+    type Store = LMDBStore;
+    type Batcher = LMDBBatch<'a>;
+
+    fn new(store: &LMDBStore) -> Result<LMDBBatch<'a>, DatastoreError> {
+        Ok(LMDBBatch {
+            db: store.curr_db.clone(),
+            tx: lmdb::WriteTransaction::new(store.env.clone())?,
+        })
+    }
+
+    fn put_raw(&mut self, key: &[u8], value: Vec<u8>) -> Result<(), DatastoreError> {
+        {
+            let mut accessor = self.tx.access();
+            accessor.put(&self.db, key, &value, lmdb::put::Flags::empty())?;
+        }
+        Ok(())
+    }
+
+    fn commit(self) -> Result<(), DatastoreError> {
+        self.tx.commit().map_err(|e| e.into())
+    }
+
+    fn abort(self) -> Result<(), DatastoreError> {
+        Ok(())
+    }
+}
+
+impl From<lmdb::error::Error> for DatastoreError {
+    fn from(err: lmdb::error::Error) -> Self {
+        let err_msg = format!("LMDB Error: {}", err.to_string());
+        DatastoreError::InternalError(err_msg)
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::{LMDBBuilder, LMDBStore};
+    use crate::keyvalue_store::{BatchWrite, DataStore, DatastoreError};
+    use crate::lmdb::LMDBBatch;
+    use bincode::{deserialize, serialize};
+    use rand::{OsRng, RngCore};
+    use std::{fs, str};
+
+    #[derive(Serialize, Deserialize, PartialEq, Debug)]
+    struct Entity {
+        x: f32,
+        y: f32,
+    }
+
+    #[derive(Serialize, Deserialize, PartialEq, Debug)]
+    struct World(Vec<Entity>);
+
+    fn to_bytes(i: u32) -> Vec<u8> {
+        i.to_le_bytes().to_vec()
```

Comment:
This is not a big issue, but to_le_bytes and from_le_bytes are considered unstable. It might be worth writing small replacement functions.